policykit/policyengine/utils.py

```diff
@@ -38,31 +53,95 @@ def construct_authorize_install_url(request, integration, community=None):
 
     # Redirect to the plugin-specific install endpoint, which will complete the setup process (ie create the SlackCommunity)
     redirect_uri = f"{settings.SERVER_URL}/{integration}/install"
-    encoded_redirect_uri = quote(redirect_uri, safe='')
+    encoded_redirect_uri = quote(redirect_uri, safe="")
 
     # store state in user's session so we can validate it later
     state = "".join([str(random.randint(0, 9)) for i in range(8)])
-    request.session['community_install_state'] = state
+    request.session["community_install_state"] = state
 
     # if not specified, metagov will create a new community and pass back the slug
-    community_slug = community.metagov_slug if community else ''
+    community_slug = community.metagov_slug if community else ""
     url = f"{settings.METAGOV_URL}/auth/{integration}/authorize?type=app&community={community_slug}&redirect_uri={encoded_redirect_uri}&state={state}"
     logger.debug(url)
     return url
 
+
 def get_starterkits_info():
     """
     Get a list of all starter-kit names and descriptions.
     """
     starterkits = []
     cur_path = os.path.abspath(os.path.dirname(__file__))
-    dir_path = os.path.join(cur_path, f'../starterkits')
+    dir_path = os.path.join(cur_path, f"../starterkits")
     for kit_file in os.listdir(dir_path):
         kit_path = os.path.join(dir_path, kit_file)
         f = open(kit_path)
         data = json.loads(f.read())
-        starterkits.append({
-            'name': data['name'],
-            'description': data['description']
-        })
+        starterkits.append({"name": data["name"], "description": data["description"]})
     return starterkits
+
+
+def initialize_starterkit_inner(community, kit_data, creator_token=None):
```

Comment:
@rickygv99 I moved the bulk of your recent starterkit changes into the `utils.py` file so we can test them! see `test_starterkits.py`